doc/api/process.md

```diff
@@ -1488,8 +1488,9 @@ Android)
 
 * {Stream}
 
-The `process.stderr` property returns a [Writable][] stream equivalent to or
-associated with `stderr` (fd `2`).
+The `process.stderr` property returns a stream equivalent to or
+associated with `stderr` (fd `2`). It is a `net.Socket` (inherits from `stream.Duplex`) 
+unless it is from/to a file, in which case it is a [Writable][] stream.
```

Comment:
@sam-github Thank you for reviewing. I made the requested changes and rebased with master. I had some merge conflicts on the same sections in the docs. 
Can you review again? Thanks!